direct/direct.go

```diff
@@ -24,10 +24,11 @@ type directClient struct {
 	con       *websocket.Conn
 	responses map[string]chan *types.Envelope
 	m         sync.Mutex
+	sub       *substrate.Substrate
 }
 
 // id is the twin id that is associated with the given identity.
-func NewClient(ctx context.Context, identity substrate.Identity, url string, id uint32, session string) (rmb.Client, error) {
+func NewClient(ctx context.Context, identity substrate.Identity, url string, id uint32, session string, sub *substrate.Substrate) (rmb.Client, error) {
```

Comment:
I think it's better instead of passing a `substrate.Substrate` client if we pass an interface of a TwinDB for example. as follows
```go
type TwinDB interface{
     GetTwin(id: uint32) Twin
}
```
The thing is the twin db can then hold instance to the substrate `Manager` and then it can do internal cashing for example if u get the same twin you don't get the object from substrate everytime

Reply:
This also means for testing you can pass a mock of the  TwinDB that return some test twin